### PR TITLE
Added an Instant NFT Minter Page

### PR DESCRIPTION
**Pull request by Saviour1001**
Added a new page for NFT minter. 
It takes Metadata and saves it to IPFS and later mints it on the testnet. 
One can simply change the contract address in the code and the same mechanism will get shifted to mainnet. 
Tested on Polygon Mumbai Testnet
image

### CODE REVIEW

**Comment by K12609** [comment]
Commit

**Comment by K12609** [comment]
Merged

**Review by BanalitoRaulito — COMMENTED** [review]
Hey, left few comments to improve

File: src/components/NFT_Minter/NFT_Minter.jsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+      <Card
+        style={styles.card}
+        bodyStyle={{ padding: "18px" }}
+        title={<div>NFT Minter</div>}
+        size="large"
+      >
+        This minter currently supports only Polygon Mumbai NFTs.
+        <br />
+        <br></br>
+        {/* <select name="selectList" id="selectList">
```
Comment (BanalitoRaulito) [inline src/components/NFT_Minter/NFT_Minter.jsx:133]: Why is this commented out?

File: src/App.jsx [hunk 2]
```diff
@@ -51,7 +57,8 @@ const styles = {
   },
 };
 const App = ({ isServerInfo }) => {
-  const { isWeb3Enabled, enableWeb3, isAuthenticated, isWeb3EnableLoading } = useMoralis();
+  const { isWeb3Enabled, enableWeb3, isAuthenticated, isWeb3EnableLoading } =
```
Comment (BanalitoRaulito) [inline src/App.jsx:60]: Try not to commit changes that are not related to the change.

File: src/App.jsx [hunk 3]
```diff
@@ -167,7 +177,13 @@ const App = ({ isServerInfo }) => {
 
 export const Logo = () => (
   <div style={{ display: "flex" }}>
-    <svg width="60" height="38" viewBox="0 0 50 38" fill="none" xmlns="http://www.w3.org/2000/svg">
+    <svg
```
Comment (BanalitoRaulito) [inline src/App.jsx:180]: Same here. Try not to commit changes that are not related to the change.

File: src/App.jsx [hunk 4]
```diff
@@ -15,6 +20,7 @@ import NativeBalance from "components/NativeBalance";
 import "./style.css";
 import QuickStart from "components/QuickStart";
 import Contract from "components/Contract/Contract";
+import CreateNFT from "./components/NFT_Minter/NFT_Minter";
```
Comment (BanalitoRaulito) [inline src/App.jsx:23]: Do not use _ in component name. 
Also I don't see why it could not be directly in components directory ("./components/mintNft")

Also keep naming consistent.
If the file name is mintNFT it should also export mintNFT.
Then you can write:
import mintNFT from "./components/mintNFT"; (which is way better)

File: src/components/NFT_Minter/NFT_Minter.jsx [hunk 5]
```diff
@@ -0,0 +1,172 @@
+import React from "react";
+import Web3 from "web3";
```
[the remark this answers is not in the document]
Comment (Saviour1001) [reply src/components/NFT_Minter/NFT_Minter.jsx]: fixed this

File: src/components/NFT_Minter/NFT_Minter.jsx [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+    justifyContent: "space-between",
+    alignItems: "center",
+  },
+};
+
+const Moralis = require("moralis");
+
+function CreateNFT() {
+  Moralis.initialize(process.env.REACT_APP_MORALIS_APPLICATION_ID);
+  Moralis.serverURL = process.env.REACT_APP_MORALIS_SERVER_URL;
```
[the remark this answers is not in the document]
Comment (Saviour1001) [reply src/components/NFT_Minter/NFT_Minter.jsx]: Fixed

File: src/components/NFT_Minter/NFT_Minter.jsx [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+  async function AddLibrary(urlOfTheLibrary) {
+    const script = document.createElement("script");
+    script.src = urlOfTheLibrary;
+    script.async = true;
+    document.body.appendChild(script);
+  }
+  AddLibrary("https://cdn.jsdelivr.net/npm/web3@latest/dist/web3.min.js");
+  AddLibrary("https://unpkg.com/moralis/dist/moralis.js");
+  const web3 = new Web3(window.ethereum);
```
[the remark this answers is not in the document]
Comment (Saviour1001) [reply src/components/NFT_Minter/NFT_Minter.jsx]: Resolved

File: src/components/NFT_Minter/NFT_Minter.jsx [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+  let nftContractAddress = "0x351bbee7C6E9268A1BF741B098448477E08A0a53"; // Make this variable
+
+  // Ethereum Rinkeby 0x0Fb6EF3505b9c52Ed39595433a21aF9B5FCc4431
+  // Polygon Mumbai 0x351bbee7C6E9268A1BF741B098448477E08A0a53
+  // BSC Testnet 0x88624DD1c725C6A95E223170fa99ddB22E1C6DDD
+
+  // choose value as per dragdrop
+
+  let selectedChain;
+  selectedChain = "MAT";
```
[the remark this answers is not in the document]
Comment (Saviour1001) [reply src/components/NFT_Minter/NFT_Minter.jsx]: I wanted to add multichain NFT Minter ... but currently, it works on Polygon Mumbai testnet.

File: src/components/NFT_Minter/NFT_Minter.jsx (hunk 1 above)
Comment (Saviour1001) [reply src/components/NFT_Minter/NFT_Minter.jsx]: Removed

File: src/App.jsx (hunk 2 above)
Comment (Saviour1001) [reply src/App.jsx:60]: Its just styling done by formatter, haven't changed any code here

File: src/App.jsx (hunk 3 above)
Comment (Saviour1001) [reply src/App.jsx:180]: Its just styling done by formatter, haven't changed any code here

Comment (BanalitoRaulito) [reply src/App.jsx:180]: Still you have changed 5 lines in commit. Maybe try to turn off formatter?
